holidays/countries/sudan.py

```diff
@@ -0,0 +1,113 @@
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAY, JUL, AUG, SEP, FRI, SAT
+from holidays.calendars.julian import JULIAN_CALENDAR
+from holidays.groups import ChristianHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Sudan(HolidayBase, ChristianHolidays, IslamicHolidays):
+    """Sudan holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Sudan>
+        * <https://web.archive.org/web/20250820070831/https://www.sudanembassy.org.uk/public-holidays/>
+        * [Christian Holidays 2011-2018](https://web.archive.org/web/20250827155208/https://evangelicalfocus.com/world/5014/christmas-celebrations-mark-progress-of-religious-freedom-in-sudan)
+    """
+
+    country = "SD"
+    default_language = "ar_SD"
+    # %s (estimated).
+    estimated_label = tr("%s (المقدرة)")
+    supported_languages = ("ar_SD", "en_US")
+    start_year = 1985
+    weekend = {FRI, SAT}
+
+    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
+        ChristianHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=SudanIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Independence Day.
+        self._add_holiday_jan_1(tr("عيد الإستقلال"))
+
+        # Christian public holidays were suspended 2011–2018 and reinstated in 2019.
+        if self._year <= 2010 or self._year >= 2019:
+            # Coptic Christmas.
+            self._add_christmas_day(tr("عيد الميلاد القبطي"), calendar=JULIAN_CALENDAR)
+
+            # Coptic Easter.
+            self._add_easter_sunday(tr("عيد الفصح القبطي"), calendar=JULIAN_CALENDAR)
+
+            # Christmas Day.
+            self._add_christmas_day(tr("يوم عيد الميلاد"))
+
+        # Islamic New Year.
+        self._add_islamic_new_year_day(tr("رأس السنة الهجرية"))
+
+        # Prophet's Birthday.
+        self._add_mawlid_day(tr("المولد النبوي الشريف"))
+
+        # Eid al-Fitr.
+        name = tr("عيد الفطر المبارك")
+        self._add_eid_al_fitr_day(name)
+        self._add_eid_al_fitr_day_two(name)
+        self._add_eid_al_fitr_day_three(name)
+
+        if self._year >= 2020:
+            self._add_islamic_calendar_holiday(
+                name, self._islamic_calendar.eid_al_fitr_dates(self._year), days_delta=+3
```

Comment:
You need to add `_add_eid_al_fitr_day_four` and `_add_eid_al_adha_day_five` into `IslamicHolidays`.